artifactory/commands/utils/interactive_questionnaire.go

```diff
@@ -0,0 +1,247 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/c-bata/go-prompt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// The interactive questionnaire works as follows:
+//	We have to provide a map of QuestionInfo which include all possible questions may be asked.
+//	1. Mandatory Questions:
+//		* We will ask all the questions in MandatoryQuestionsKeys list one after the other.
+//	2. Optional questions:
+//		* we have to provide a slice of prompt.Suggest, in which each suggest.Text is a key of a question in the map.
+//		* after a suggest was chosen from the list, the corresponding question from the map will be asked.
+//		* aach answer is written to to the configMap using its writer, under the MapKey specified in the questionInfo.
+//		* we will execute the previous step until the WriteAndExit string was inserted.
+type InteractiveQuestionnaire struct {
+	QuestionsMap           map[string]QuestionInfo
+	MandatoryQuestionsKeys []string
+	OptionalKeysSuggests   []prompt.Suggest
+	AnswersMap             map[string]string
+}
+
+// Each question can have the following properties:
+// 		* Msg - will be printed in separate line
+// 		* PromptPrefix - will be printed before the input cursor in the answer line
+// 		* Options - In case the answer must be selected from a predefined list
+// 		* AllowVars - a flag indicates whether a variable (in form of ${var}) is an acceptable answer despite the predefined list
+// 		* Writer - how to write the answer to the final config map
+// 		* MapKey - the key under which the answer will be written to the configMap
+// 		* Callback - optional function can be executed after the answer was inserted. Can be used to implement some dependencies between questions.
+type AnswerWriter func(resultMap *map[string]interface{}, key, value string) error
+type questionCallback func(*InteractiveQuestionnaire, string) (string, error)
+
+type QuestionInfo struct {
+	Msg          string
+	PromptPrefix string
+	Options      []prompt.Suggest
+	AllowVars    bool
+	Writer       AnswerWriter
+	MapKey       string
+	Callback     questionCallback
+}
+
+const (
+	PressTabMsg      = " (press Tab for options):"
+	InvalidAnswerMsg = "Invalid answer. Please select value from the suggestions list."
+	VariableUseMsg   = " You may use dynamic variable in the form of ${key}."
+	EmptyValueMsg    = "The value can not be empty. Please enter a valid value:"
+	OptionalKey      = "OptionalKey"
+	WriteAndExist    = ":x"
```

Comment:
I think it should be **WriteAndExit**. I actually suggest renaming it to **SaveAndExit**.